File: mvp_app/MyProject/libtsparser/src/main/java/com/tosmart/xiexuming/libtsparser/section/ProgramAssociationSection.java
package com.tosmart.xiexuming.libtsparser.section;

import androidx.annotation.NonNull;

import java.util.Arrays;

/**
 * @author xiexuming
 */
public class ProgramAssociationSection extends Section {
    private int transportStreamId;
    private int versionNumber;
    private int currentNextIndicator;
    private int sectionNumber;
    private int lastSectionNumber;
    private byte[] crcCode;

    public void setHeadDate(int[] programAssociationSectionHead) {
        setTableId(programAssociationSectionHead[0]);
        setReserved(programAssociationSectionHead[1]);
        setSectionLength(programAssociationSectionHead[2]);
        transportStreamId = programAssociationSectionHead[3];
        versionNumber = programAssociationSectionHead[4];
        currentNextIndicator = programAssociationSectionHead[5];
        sectionNumber = programAssociationSectionHead[6];
        lastSectionNumber = programAssociationSectionHead[7];
    }

    @NonNull
    @Override
    public String toString() {
        return "ProgramAssociationSection{" +
                "tableId=" + getTableId() +
                ", sectionLength=" + getSectionLength() +
                ", sectionSyntaxIndicator=" + getSectionSyntaxIndicator() +
                ", reserved=" + getReserved() +
                "transportStreamId=" + transportStreamId +
                ", versionNumber=" + versionNumber +
                ", currentNextIndicator=" + currentNextIndicator +
                ", sectionNumber=" + sectionNumber +
                ", lastSectionNumber=" + lastSectionNumber +
                ", crcCode=" + crcCode[0] + " " + crcCode[1] + " " + crcCode[2] + " " + crcCode[3] +
                ", data=" + Arrays.toString(getData()) +
                '}';
    }

    @Override
    public void setData(byte[] data) {
        super.setData(data);
    }

    private void setProgramContent(byte[] programContent) {
        setData(programContent);
    }

    public int getTransportStreamId() {
        return transportStreamId;
    }

    public void setTransportStreamId(int transportStreamId) {
        this.transportStreamId = transportStreamId;
    }

    public int getVersionNumber() {
        return versionNumber;
    }

    public void setVersionNumber(int versionNumber) {
        this.versionNumber = versionNumber;
    }

    public int getCurrentNextIndicator() {
        return currentNextIndicator;
    }

    public void setCurrentNextIndicator(int currentNextIndicator) {
        this.currentNextIndicator = currentNextIndicator;
    }

    public int getSectionNumber() {
        return sectionNumber;
    }

    public void setSectionNumber(int sectionNumber) {
        this.sectionNumber = sectionNumber;
    }

    public int getLastSectionNumber() {
        return lastSectionNumber;
    }

    public void setLastSectionNumber(int lastSectionNumber) {
        this.lastSectionNumber = lastSectionNumber;
    }

    public byte[] getCrcCode() {
        return crcCode;
    }

    public void setCrcCode(byte[] crcCode) {
        this.crcCode = crcCode;
    }
}

File: newpaserapp/libtosmartparse/src/main/java/com/example/libtosmartparse/section/ProgramAssociationSection.kt
package com.example.libtosmartparse.section

import java.util.*

/**
 * @author xxm
 * 2022/7/17
 **/
class ProgramAssociationSection : Section() {
    private var transportStreamId = 0
    private var versionNumber = 0
    private var currentNextIndicator = 0
    private var sectionNumber = 0
    private var lastSectionNumber = 0
    private lateinit var crcCode: ByteArray

    fun setHeadDate(programAssociationSectionHead: IntArray) {
        setTableId(programAssociationSectionHead[0])
        setReserved(programAssociationSectionHead[1])
        setSectionLength(programAssociationSectionHead[2])
        transportStreamId = programAssociationSectionHead[3]
        versionNumber = programAssociationSectionHead[4]
        currentNextIndicator = programAssociationSectionHead[5]
        sectionNumber = programAssociationSectionHead[6]
        lastSectionNumber = programAssociationSectionHead[7]
    }

    override fun toString(): String {
        return "ProgramAssociationSection{" +
                "tableId=" + getTableId() +
                ", sectionLength=" + getSectionLength() +
                ", sectionSyntaxIndicator=" + getSectionSyntaxIndicator() +
                ", reserved=" + getReserved() +
                "transportStreamId=" + transportStreamId +
                ", versionNumber=" + versionNumber +
                ", currentNextIndicator=" + currentNextIndicator +
                ", sectionNumber=" + sectionNumber +
                ", lastSectionNumber=" + lastSectionNumber +
                ", crcCode=" + crcCode[0] + " " + crcCode[1] + " " + crcCode[2] + " " + crcCode[3] +
                ", data=" + Arrays.toString(getData()) +
                '}'
    }

    private fun setProgramContent(programContent: ByteArray) {
        setData(programContent)
    }

    fun getTransportStreamId(): Int {
        return transportStreamId
    }

    fun setTransportStreamId(transportStreamId: Int) {
        this.transportStreamId = transportStreamId
    }

    fun getVersionNumber(): Int {
        return versionNumber
    }

    fun setVersionNumber(versionNumber: Int) {
        this.versionNumber = versionNumber
    }

    fun getCurrentNextIndicator(): Int {
        return currentNextIndicator
    }

    fun setCurrentNextIndicator(currentNextIndicator: Int) {
        this.currentNextIndicator = currentNextIndicator
    }

    fun getSectionNumber(): Int {
        return sectionNumber
    }

    fun setSectionNumber(sectionNumber: Int) {
        this.sectionNumber = sectionNumber
    }

    fun getLastSectionNumber(): Int {
        return lastSectionNumber
    }

    fun setLastSectionNumber(lastSectionNumber: Int) {
        this.lastSectionNumber = lastSectionNumber
    }

    fun getCrcCode(): ByteArray {
        return crcCode
    }

    fun setCrcCode(crcCode: ByteArray) {
        this.crcCode = crcCode
    }
}
